registration/management/commands/create_users.py

```diff
@@ -0,0 +1,81 @@
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand, CommandError
+from participant.models import Troop
+
+import csv
+import os
+
+
+class Command(BaseCommand):
+    help = 'Imports a set of users from a CSV file'
+
+    def add_arguments(self, parser):
+        parser.add_argument('filename')
+
+    def handle(self, *args, **options):
+        try:
+            with open(options['filename'], newline='') as csv_file:
+                csv_reader = csv.reader(csv_file, delimiter=';')
+                columns = self._extract_columns(next(csv_reader))
+
+                lines = []
+
+                for i, row in enumerate(csv_reader):
+                    try:
+                        line = {
+                            'troop_number': row[columns['troop_number']],
+                            'troop_name': row[columns['troop_name']],
+                            'email': row[columns['email']],
+                            'firstname': row[columns['firstname']],
+                            'lastname': row[columns['lastname']],
+                            'line_number': i+2,
+                        }
+
+                        lines.append(line)
+
+                    except Exception as e:
+                        self.stderr.write('Line {}: {}'.format(i+2, e.__cause__))
+
+                self.import_lines(lines)
+
+        except FileNotFoundError:
+            raise CommandError('File {} not found'.format(os.path.abspath(options['filename'])))
+
+    def import_lines(self, lines):
+        created_users, created_troops = 0, 0
+
+        for line in lines:
+            try:
+                troop, created = Troop.objects.get_or_create(
+                    number=line['troop_number'],
+                    name=line['troop_name'],
+                )
+
+                if created:
+                    created_troops += 1
+
+                user, created = get_user_model().objects.get_or_create(
+                    email=line['email'],
+                    first_name=line['firstname'],
```

Comment:
`first_name` and `last_name` should be in a [`defaults`](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#django.db.models.query.QuerySet.get_or_create) dict (otherwise there could be a unicity error in case a user is inserted mutliple times with the ame email, but there is a typo in one firstname)

Reply:
A `User` cannot be inserted multiple times with the same email address. `email` has to unique so I think this is fine.

Reply:
It will trigger an exception if the imported file is incoherent:

```
john@example.com,John,Doe;
john@example.com,John Jr,Doe;
```

=> since it won't find `John Jr`, it will try to insert it and the unique constraint will raise an exception.

Using `defaults` will be more robust (I admit this is an edge case, but we want a robust software :-)

Reply:
Yes, and having that Exception is good here because otherwise we wouldn't notice that there is something off. What if the first record was wrong and the second correct? The script would create the wrong user.

Reply:
good point